entity/src/main/java/com/wilma/entity/users/UserAccount.java

```diff
@@ -0,0 +1,54 @@
+
+package com.wilma.entity.users;
+
+import lombok.*;
+
+import javax.persistence.*;
+import java.util.Set;
+
+/**
+ * User super class
+ */
+@Getter
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "user")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+public class UserAccount {
+
+    @Id
+    @GeneratedValue
+    @Column(name = "id", nullable = false)
+    private Integer userId;
+
+    @Column(name = "username")
+    private String username;
+
+    @Column(name = "password")
+    private String password;
+
+    @Column(name = "email")
+    private String email;
+
+    @Column(name = "credentials_non_expired")
+    public boolean credentialsNonExpired;
+
+    @Column(name = "account_non_locked")
+    public boolean accountNonLocked;
+
+    @Column(name = "account_non_expired")
+    public boolean accountNonExpired;
+
+    @Column(name = "active")
+    private boolean enabled;
+
+    @ManyToMany(fetch = FetchType.EAGER)
+    @JoinTable(name = "user_roles",
+            joinColumns = @JoinColumn(name = "userId", referencedColumnName = "id"),
+            inverseJoinColumns = @JoinColumn(name = "roleId", referencedColumnName = "id"))
+    private Set<Role> roles;
```

Comment:
SQL will ignore upper/lower case, so `userId` will become `userid` (or `USERID`)
```suggestion
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "user_roles",
            joinColumns = @JoinColumn(name = "userId", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name = "roleId", referencedColumnName = "id"))
    private Set<Role> roles;
```